pymess/management/commands/send_messages_batch.py

```diff
@@ -69,17 +62,15 @@ def _print_result(self, title, message_pks):
             self.stdout.write('{}: {}'.format(title, len(message_pks)))
 
     def handle(self, type, *args, **options):
-        sender = self.senders[type]()
-
-        if not sender.is_turned_on_batch_sending():
-            raise CommandError('Batch sending is turned off')
-
-        try:
-            for _ in range(sender.get_batch_size()):
-                if not self._send_message(sender):
-                    break
-            self._print_result('sent messages', self.send_message_pks)
-            self._print_result('failed messages', self.failed_message_pks)
-        except DatabaseError:
-            self.stdout.write('messages are already locked')
-
+        for sender in {self.senders[type](), self.senders[type](is_testing=True)}:
```

Comment:
so message is created without backend and backend is used only for sending message? 

So:
* batch_size cannot be defined in the backend (more backends can be used during sending messages)
* backend cannot manage if batch sending is turned on or off (all the backends must have same settings)

Reply:
you must get backend per recipient here too

Reply:
For my opinion there must be controller for every type of message (email, sms, dialer,...) Controler will manage creating message, batch sending, etc. And backend only manage sending the message to the service. So you must split backend to two classes now

Reply:
ok, I will refactor it

Reply:
I have refactored it. Only `is_turned_on_batch_sending` is still in the backend (because it's being accessed when the message is being send within the backend. I didn't want to do backend -> controller relation, there's only controller -> backend